### ISSUE TITLE
web console alignment and behavior isn't quite right compared to X11/windows. 

### ISSUE THREAD

**Issue by wschaub** [kej715/DtCyber#129]
**Describe the bug**
Character spacing isn't quite right on the web console.

I think the DS1 tests should be run to dial in the web console support to be in line with X11 console in terms of alignment
and other behavior that fails the DS1 diagnostic tests. 

**To Reproduce**
1. Start NOS 2.8.7 
2. type X.DS1 on the console
3. type Z L S S
4. Compare the same test patten with the X11 display. 

**Expected behavior**
Alignment should match the DS1 diagnostic tests

**Screenshots**
X11 DS1 test Z dot/chat centering comparison:
![image](https://github.com/kej715/DtCyber/assets/955669/90e507cc-d588-484f-8820-370cef3085de)

Web console DS1 test Z dot/char centering comparison:
The display is on the incorrect tube it's on the right side, the character size is wrong and the alignment is incorrect. 
![image](https://github.com/kej715/DtCyber/assets/955669/69743079-9bbf-478a-916d-dafd138a0e54)

**Comment by kej715** [comment]
Code has been committed to address this issue. In particular, issues with screen selection and font size have been corrected, and DS1 confirms that those issues have been resolved. However, the X11 display might not be correct. Specifically, the X should probably be centered over the cross. The programming example in 60333900B_6602_6612_Console_Display_Sep74.pdf on BitSavers appears to expect that the lower left corner of a character is displayed at the specified display coordinate.

**Comment by wschaub** [comment]
> Code has been committed to address this issue. In particular, issues with screen selection and font size have been corrected, and DS1 confirms that those issues have been resolved. However, the X11 display might not be correct. Specifically, the X should probably be centered over the cross. The programming example in 60333900B_6602_6612_Console_Display_Sep74.pdf on BitSavers appears to expect that the lower left corner of a character is displayed at the specified display coordinate.

The character spacing still seems a bit off.  see the spacing in the "DISK BUSY" message on the deadstart screen.

![image](https://github.com/kej715/DtCyber/assets/955669/aec42db3-7504-4bb6-96f0-f185024b8e98)

**Comment by kej715** [comment]
It's a characteristic of the font, which appears to left-justify letters in
the pixel block.

**Comment by kej715** [comment]
The current code uses font family Lucida Typewriter (same as X11 console
uses). It draws each character such that it's lower left corner is at the
specified console X/Y coordinate. A basic assumption is that the font used
has fixed width and each character of the font will be centered within the
pixel block. If the actual font used at runtime doesn't conform to that
assumption, then characters might appear to be "off center".

**Comment by wschaub** [comment]
Are you saying this is a problem with my local system then?  I'm not sure how I would fix this. I just load the console interface up in chrome and I certainly have the 75dpi X11 font that the X11 console wants to use, since I run that all the time.

**Comment by kej715** [comment]
No, it's probably not a problem with your local system. It appears to be a
problem with the HTML version (not the X-11 version) of the Lucida
Typewriter font family definition. It is apparently not as fixed-width as
expected, or the "I" character is not centered within it. For example, the
"I" character appears to have a smaller width than "D", "S", or "K"
characters. Consequently, it has the appearance of being "off center" in
its character block. We could try using other font families, but Lucida
Typewriter was chosen in X-11 due to its resemblance to the historic
console font.

The code for drawing text is very straightforward. Basically, draw a
character at the desired coordinates, then increase the X coordinate by the
font width, then draw the next character. If a character isn't centered
within the character block (i.e., x/y width of each fixed-size character),
then it seems that the only way this can happen is if the font definition
itself does not have the character centered, or the font isn't truly
fixed-width.

I just tried changing the font family from "Lucida Typewriter" to the
generic name "monospace", as an experiment. The character spacing became
uniform, but the appearance was very different from the historic font, and
it was somewhat difficult to read.

**Comment by kej715** [comment]
P.S.: You can easily experiment yourself. For example, look for "Lucida
Typewriter" in webterm/www/js/console.js, and change it to the name of
another font family, e.g.,. "monospace". If you think that "monospace" is a
better choice (i.e., it's better to have truly uniform spacing than to be
closer in appearance to historical font), then I'll be happy to commit that
change to the repo.

**Comment by philclaridge** [comment]
FYI, I am working on a next generation web character generator / font for a accurate rendering of CDC character set for my console emulation (totally different from the text based console I added to the distribution over Xmas). 

My goal is simple - I want a CDC6600 running as a pure demon with support for both a remote console and remote dtcyber command line interface. 

My console vector display will scale down for an LCD (with simplified character display) and up to emulate a full size vector drawn ’scope tube. I am aiming for a pixel (vector) perfect display.

Happy to chat to any of you about some of the different design options I have been evaluating (all my work is currently in a private branch). 

Would be nice not to have too many versions of remote cyber display protocols. 

My background is electronics so like others before me I’ll be returning to the beam deflection, step timing, and phosphor physics for the larger display sizes, from the console hardware design I can see why the characters are shapes are as shown.

Aside: In longer term I would like to move the local console from X11 to SDL2 as well.

Phil
